packages/bpk-component-banner-alert/src/BpkBannerAlert.js

```diff
@@ -79,42 +82,98 @@ const ToggleButton = (props) => {
   );
 };
 
-ToggleButton.propTypes = {
+ExpandToggleButton.propTypes = {
   label: PropTypes.string.isRequired,
   expanded: PropTypes.bool.isRequired,
 };
 
+const DismissButton = (props) => {
+  const classNames = [getClassName('bpk-banner-alert__expand-icon')];
+  const { label, ...rest } = props;
+
+  return (
+    <button
+      className={getClassName('bpk-banner-alert__toggle-button')}
+      aria-label={label}
+      title={label}
+      {...rest}
+    >
+      <CloseIcon
+        className={classNames.join(' ')}
+      />
+    </button>
+  );
+};
+
+DismissButton.propTypes = {
+  label: PropTypes.string.isRequired,
+};
+
 class BpkBannerAlert extends Component {
   constructor(props) {
     super(props);
 
     this.state = {
       expanded: false,
+      shown: !this.props.fadeIn,
     };
 
-    this.onExpand = this.onExpand.bind(this);
+    this.onToggleExpand = this.onToggleExpand.bind(this);
+    this.onDismiss = this.onDismiss.bind(this);
   }
 
-  onExpand() {
+  componentDidMount() {
+    this.onShown();
+  }
+
+  onToggleExpand() {
     this.setState(state => ({
       expanded: !state.expanded,
     }));
+    if (this.props.onAction !== null) {
+      this.props.onAction();
```

Comment:
for consistency's sake if you're using the `typeof  === 'function'` check, use it everywhere there is a similar call